flight/PiOS/Common/pios_mpu6000.c

```diff
@@ -575,14 +614,60 @@ static void PIOS_MPU6000_Task(void *parameters)
 		    BUFFER_SIZE,
 		};
 
-		uint8_t mpu6000_send_buf[BUFFER_SIZE] = { PIOS_MPU60X0_ACCEL_X_OUT_MSB | 0x80, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
-		uint8_t mpu6000_rec_buf[BUFFER_SIZE];
+		enum {
+			IDX2_SPI_DUMMY_BYTE = 0,
+			IDX2_GYRO_XOUT_H,
+			IDX2_GYRO_XOUT_L,
+			IDX2_GYRO_YOUT_H,
+			IDX2_GYRO_YOUT_L,
+			IDX2_GYRO_ZOUT_H,
+			IDX2_GYRO_ZOUT_L,
+			BUFFER2_SIZE,
+		};
+
+
+		static bool AllSamplingFlg = false; // flag that indicates that all data should be sampled
+		static bool DoOutputFlg    = false;   // flag that indicates that we have to compute an new output at the down sampled frequency
+
+		// buffer for sampling all data
+		uint8_t mpu6000_send_all_buf[BUFFER_SIZE] = { PIOS_MPU60X0_ACCEL_X_OUT_MSB | 0x80, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
```

Comment:
Had overseen that. 
I looked to often to the code, and that does not seemed like a critical part for me, so I think my brain filtered that.

Thanks for the hint.